src/components/Tabs/Tabs.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled, { css } from 'styled-components';
+
+const TabsDiv = styled.div`
+  text-transform: uppercase;
+  font-weight: bold;
+  user-select: none;
+  -webkit-user-select: none;
+  -moz-user-select: none;
+  -ms-user-select: none;
+  ${({ vertical }) => {
+    if (vertical) {
+      return css`  
+    width: 25%;
+    display: table-cell;
+    margin-bottom: -1px;
+    text-align: left;`;
+    }
+    return css`  
+    display: flex;
+    flex-direction: row;
+    align-items: stretch;
+    flex: 1 1 auto;
+    text-align: center;
+    margin-bottom: -1px;
+    min-height: 53px;
+    text-align: center;`;
+  }}`
+  ;
+
+export default class Tabs extends React.Component {
+
+  static propTypes = {
+    /**
+     * Tab elements within container
+     */
+    children: PropTypes.node.isRequired,
+    /**
+     * The name of the tabs container
+     */
+    name: PropTypes.string,
+    /**
+     * Called when user selects the tab
+     */
+    handleSelect: PropTypes.func,
+    /**
+     * The currently selected tab
+     */
+    selectedTabIndex: PropTypes.number,
+    /**
+     * Enable vertical tab layout
+     */
+    vertical: PropTypes.bool,
+  };
+
+  renderTabs = (children, { handleSelect, selectedTabIndex, name, vertical }) => {
+    if (typeof children !== 'object') {
+      return children;
+    }
+
+    return React.Children.map(children, (child, index) => {
+      if (!React.isValidElement(child)) {
+        return child;
+      }
+
+      // rendering Tabs
+      return React.cloneElement(child, {
+        handleSelect,
+        index,
+        active: index === selectedTabIndex,
+        namespace: name,
+        vertical,
+      });
+    });
+  };
+
+  render() {
+    const tabs = this.renderTabs(this.props.children,
+      { handleSelect: this.props.handleSelect,
+        selectedTabIndex: this.props.selectedTabIndex,
+        name: this.props.name,
+        vertical: this.props.vertical,
+      });
+    return (
+      <TabsDiv vertical={this.props.vertical}>{tabs}</TabsDiv>
+    );
+  }
+}
+
+Tabs.defaultProps = {
+  vertical: false,
+};
+
+Tabs.usage = `
```

Comment:
`.usage` is deprecated; with our new documentation you include a `.md` file next to your component (like `Tabs.md`)

Reply:
got it